File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-fortran-skills/tsvc_2_s316__submitted.f90
subroutine tsvc_2_s316_fp64(a, result, len_1d) bind(C)
  use iso_c_binding, only: c_int64_t, c_double
  use omp_lib
  implicit none
  integer(c_int64_t), value, intent(in) :: len_1d
  real(c_double), intent(in) :: a(len_1d)
  real(c_double), intent(out) :: result(1)
  real(c_double) :: x
  integer :: i, n

  n = int(len_1d)
  x = a(1)
  !$omp parallel do simd reduction(min:x) schedule(static)
  do i = 2, n
    x = min(x, a(i))
  end do
  !$omp end parallel do simd
  result(1) = x
end subroutine tsvc_2_s316_fp64

File: paper_artifacts/experiments/llr9/artifacts/sources/oss120b-c/tsvc_2_s316__submitted.c
#include <float.h>
#include <stdint.h>
#include <omp.h>

// Kernel for TSVC 2, kernel s316, double-precision (fp64).
// Expected entry point: void tsvc_2_s316_fp64(double *a, double *result,
//                        int64_t LEN_1D, uint8_t *workspace, int64_t workspace_bytes);
// The workspace arguments are unused for this kernel.

void tsvc_2_s316_fp64(double *a, double *result, int64_t LEN_1D, uint8_t *workspace, int64_t workspace_bytes) {
    // Find the minimum element in a[0..LEN_1D-1] and store it in result[0].
    // Use parallel reduction to improve performance on large inputs.
    double min_val = DBL_MAX;
    #pragma omp parallel for reduction(min:min_val) schedule(static)
    for (int64_t i = 0; i < LEN_1D; ++i) {
        double v = a[i];
        if (v < min_val) {
            min_val = v;
        }
    }
    result[0] = min_val;
    // workspace and workspace_bytes are unused; they are present for ABI compatibility.
    (void)workspace;
    (void)workspace_bytes;
}

